demo/primefaces-users-portlet/src/main/java/com/liferay/faces/demos/service/CompanyLocalServiceTracker.java

```diff
@@ -25,7 +25,28 @@
  */
 public class CompanyLocalServiceTracker extends ServiceTracker<CompanyLocalService, CompanyLocalService> {
 
-	public CompanyLocalServiceTracker(BundleContext bundleContext) {
+	private BundleContext _bundleContext;
+	private CompanyLocalService _companyLocalService;
+	
+
+	public CompanyLocalServiceTracker(BundleContext bundleContext) {		
 		super(bundleContext, CompanyLocalService.class, null);
+		_bundleContext = bundleContext;
+	}
+
+	@Override
+	public CompanyLocalService addingService(ServiceReference<CompanyLocalService> reference) {
+		_companyLocalService = _bundleContext.getService(reference);
+		return _companyLocalService;
+	}
+
+	@Override
+	public void removedService(ServiceReference<CompanyLocalService> reference, CompanyLocalService service) {
+		_bundleContext.ungetService(reference);
+		_companyLocalService = null;
+	}
+	
+	public CompanyLocalService getCompanyLocalService() {
+		return _companyLocalService;
```

Comment:
@juangon I don't see how this provides anything that the `super.getService` method doesn't already do?

Reply:
Hi @ngriffin7a . If you remember we talked about the actual ServiceTracker.getService method (http://grepcode.com/file/repo1.maven.org/maven2/org.apache.felix/org.apache.felix.framework/3.0.4/org/osgi/util/tracker/ServiceTracker.java#ServiceTracker.getService%28%29). There you can see there is a "cachedService" variable which isn't updated when service is being modified or removed. That's why we need to refresh wherever reference we are using for UserLocalService or CompanyLocalService.

Reply:
So, to summarize, we can't guarantee ServiceTracker.getService will have a fresh service. If it has been called once, it will cache the previous service. That's why we should store each fresh service in a variable.